File: backend/src/main/java/com/example/skilltracker/stats/SkillDetailsStatsDto.java
package com.example.skilltracker.stats;

import java.util.List;

public record SkillDetailsStatsDto(Long skillId,
                                   String skillName,
                                   long totalMinutes,
                                   long sessionsCount,
                                   Double averageDifficulty,
                                   List<SkillDailyStatsDto> byDay) {
}

File: android/app/src/main/java/com/example/skilltracker/data/dto/SkillDetailsStatsDto.kt
package com.example.skilltracker.data.dto

data class SkillDetailsStatsDto(
    val skillId: Long,
    val skillName: String,
    val totalMinutes: Long,
    val sessionsCount: Long,
    val averageDifficulty: Double?,
    val byDay: List<SkillDailyStatsDto>
)
